app/src/main/java/fragments/DescriptionBoxFragment.java

```diff
@@ -90,7 +91,11 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         spinnerBeach = view.findViewById(R.id.spinnerBeach);
         spinnerBeach.setOnItemSelectedListener(this);
 
-        QueryUtils.queryBeachesForSpinner(spinnerBeach, view);
+        if (InternetUtil.isInternetConnected()) {
+            QueryUtils.queryBeachesForSpinner(spinnerBeach, view);
+        } else {
+            QueryUtils.queryBeachesforSpinnerOffline(spinnerBeach, view);
+        }
```

Comment:
Yeah, I'd move the Offline/not logic into the QueryUtils. Just have QueryUtils.queryBeachesForSpinner() { if (InternetUtil.isInternetConnected()) {
            QueryUtils.queryBeachesForSpinnerOnline(spinnerBeach, view);
        } else {
            QueryUtils.queryBeachesforSpinnerOffline(spinnerBeach, view);
        }
}